### ISSUE TITLE
Rule for only one jsx react element per line?

### ISSUE THREAD

**Issue by TSMMark** [yannickcr/eslint-plugin-react#1491]
I would like to disallow multiple elements/components being opened on one line:

```jsx
<Outer><Inner/></Outer>
```

and fix to:

```jsx
<Outer>
  <Inner/>
</Outer>
```

I searched all over and could not find something like this. Thank you.

**Comment by ljharb** [comment]
Would this rule also warn on `<A /><B />` and fix to:
```jsx
<A />
<B />
```
?

**Comment by TSMMark** [comment]
@ljharb Yes indeed.

However, maybe it could be configurable to conditionally support either of the following variants:

```jsx
<A>{whatever()}</A>
```

```jsx
<A>
  {whatever()}
</A>
```

**Comment by ljharb** [comment]
Sounds great; want to make a PR? :-D

**Comment by TSMMark** [comment]
@ljharb I would be happy to when I get the chance. Never worked in this codebase before, but I assume [`jsx-max-props-per-line.js`](https://github.com/yannickcr/eslint-plugin-react/blob/master/lib/rules/jsx-max-props-per-line.js) is probably a good starting reference point.

**Comment by TSMMark** [comment]
@ljharb Done in #1497

**Comment by TSMMark** [comment]
It's merged. Closing
